WifiART/src/test/userinterface/ZTest.java

```diff
@@ -20,54 +20,387 @@
+        STATE_DEFAULT.getFloor(),
+        STATE_DEFAULT.getHeightAboveFloorMeters(),
+        STATE_DEFAULT.getHeightAboveFloorUncertaintyMeters(),
+        ExpectedToMove.NOT_EXPECTED_TO_MOVE
+    );
+    private static final String BUFFER_WITH_FIXED_LOCATION = "0406000000000000";
+
+    // Values set to default, with variable location.
+    private static final ZState STATE_WITH_VARIABLE_LOCATION = buildZState(
+        STATE_DEFAULT.getFloor(),
+        STATE_DEFAULT.getHeightAboveFloorMeters(),
+        STATE_DEFAULT.getHeightAboveFloorUncertaintyMeters(),
+        ExpectedToMove.EXPECTED_TO_MOVE
+    );
+    private static final String BUFFER_WITH_VARIABLE_LOCATION = "0406010000000000";
+
+
     private final ZModel model = new ZModel(new ZState());
 
+
     /**
-     * Test the encoding with a stationary STA, floor number = 4, height of 2.8m above the floor,
-     *  with height uncertainty of 0.1m. New Android version.
+     * Constructs a ZState with pre-determined parameter values.
+     *
+     * @param floor the floor number
+     * @param heightAboveFloorMeters the height above the floor (in meters)
+     * @param heightAboveFloorUncertaintyMeters the uncertainty for the height (in meters)
+     * @param expectedToMove the enum representing the STA location movement pattern
+     * @return the ZState with the paramter values set.
+     */
+    private static ZState buildZState(int floor,
+                                          double heightAboveFloorMeters,
+                                          double heightAboveFloorUncertaintyMeters,
+                                          ExpectedToMove expectedToMove) {
+        ZState state = new ZState();
+        state.setFloor(floor);
+        state.setHeightAboveFloorMeters(heightAboveFloorMeters);
+        state.setHeightAboveFloorUncertaintyMeters(heightAboveFloorUncertaintyMeters);
+        state.setExpectedToMove(expectedToMove);
+        return state;
+    }
+
+    /**
+     * Test setting the the Floor parameter to the maximum value.
+     */
+    @Test
+    void testSetMaxFloor() {
+        ZState state = new ZState();
+
+        state.setFloor(MAX_FLOOR_NUMBER);
+
+        assertEquals(MAX_FLOOR_NUMBER, state.getFloor());
+    }
+
+    /**
+     * Test setting the the Floor parameter to the minimum value.
+     */
+    @Test
+    void testSetMinFloor() {
+        ZState state = new ZState();
+
+        state.setFloor(MIN_FLOOR_NUMBER);
+
+        assertEquals(MIN_FLOOR_NUMBER, state.getFloor());
+    }
+
+    /**
+     * Test setting the the Height Above Floor parameter to the maximum value.
+     */
+    @Test
+    void testSetMaxHeightAboveFloor() {
+        ZState state = new ZState();
+
+        state.setHeightAboveFloorMeters(MAX_HEIGHT_ABOVE_FLOOR_METERS);
+
+        assertEquals(MAX_HEIGHT_ABOVE_FLOOR_METERS, state.getHeightAboveFloorMeters());
+    }
+
+    /**
+     * Test setting the the Height Above Floor parameter to the minimum value.
```

Comment:
duplicate word

Reply:
Good catch. Turns out "the" was duplicated for several javadoc comments here.